packages/extension-chakra-storefront/src/components/icons/index.test.js

```diff
@@ -5,34 +5,33 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import React from 'react'
-import {useIntl} from 'react-intl'
-import {within} from '@testing-library/dom'
-import {render} from '@testing-library/react'
-import {renderWithProviders} from '../../utils/test-utils'
+import {defineMessage} from 'react-intl'
+import {renderWithChakraProvider, renderWithProviders} from '../../utils/test-utils'
 import * as Icons from '../../components/icons/index'
-
-jest.mock('react-intl', () => ({
-    ...jest.requireActual('react-intl'),
-    useIntl: jest.fn()
-}))
-
+import {icon} from './index'
 beforeEach(() => {
     jest.clearAllMocks()
 })
 
 test('renders svg icons with Chakra Icon component', () => {
     renderWithProviders(<Icons.CheckIcon />)
     const svg = document.querySelector('.chakra-icon')
-    const use = within(svg).getByRole('presentation')
+
     expect(svg).toBeInTheDocument()
+    const use = svg.querySelector('use')
     expect(use).toBeInTheDocument()
     expect(svg).toHaveAttribute('viewBox', '0 0 24 24')
     expect(use).toHaveAttribute('xlink:href', '#check')
 })
 
-test('uses intl from context when rendered with providers', () => {
+test('renders icon with correct aria attributes (with localed message)', () => {
```

Comment:
The old did not make sense in the first place. In the code comment said avoid using `useIntl` but we tried to check for `useIntl` call during the test.

It is not a good practice to test the component implementation with @testing-library/react. It is recommended to test the component behavior and output we expect to see.

In this case, we want to ensure that the Icon is render with correct attributes that we pass in with correct localized msg. Hence, we asserted the outcome after rendering the Icon.